### ISSUE TITLE
whether RERconverge has to be used for converge ?

### ISSUE THREAD

**Issue by zhxh233** [nclark-lab/RERconverge#92]
Hi,

I think the basic principle of RERconverge is to calculate the RERs of each gene and then make a Kendall's tau test (if it's a binary phenotype). If I'm correct, for any given set of species (in a tree), I can calculate the RERs of each gene, to test whether there are some fast-evolving genes in some species. After that, if there are some species come from different ancestors, I can say there is a convergent signal. But if there is only one species (or several species share a same ancestor), I can still assume that these genes are fast-evolving genes based on the RERs, right?
I got this but I didn't find any reference to support it, please let me know if I'm right.

Regards,
XH

**Comment by zhxh233** [comment]
And there is another question when I used *getAllResiduals* to get RERs. In my results, the result table got many columns with the same species name just like this
![图片](https://github.com/nclark-lab/RERconverge/assets/57990205/7a750ab4-d79a-4729-99d7-b5417230a085)
In another paper using RERconverge(Richardson, R. et al. The genomic basis of temporal niche evolution in a diurnal rodent. Current Biology 33, 3289-3298.e6 (2023).), the same goes for their results.
![图片](https://github.com/nclark-lab/RERconverge/assets/57990205/009ba485-cfc9-4422-a71d-62f418343422)
So if I need to use RERs, which columns should I use ? Or replace *NA* with 0, and use the sum of all values of the same species?

Thank you!

**Comment by nclark-lab** [comment]
Yes, the general statements you make are correct and are supported by Chikina Robinson and Clark. MBE 2016 and Partha Kowalczyk Clark and Chikina. MBE 2019.
For the RER matrix, the repeated column names are there because there are many possible "paths" through the species tree that appear when one or many species are missing for a given gene. All possible paths are represented in the columns of the table. You should use the non-zero value for a given gene.
